### PR TITLE
add .gitignore file 

### PR DESCRIPTION
**Pull request by garciagenrique**

### CODE REVIEW

**Review by maxnoe — CHANGES_REQUESTED** [review]
Editor and OS specific things should go into your user-wide gitignore, not into the projects gitignore. Otherwise every project would need to maintain a gitignore for each and every editor in use.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Packages/installer info
+*.egg
+*.eggs
+*.egg-info
+dist
+build
+_version.py
+
+# Mac OSX
```
Comment (maxnoe) [inline .gitignore:14]: remove macos and pycharm specific things

Comment (vuillaut) [reply .gitignore:14]: I am not sure I agree here.
E.g. having `.DS_Store` in the `.gitignore` helps all users on macos while it does not hurt anyone else.
Once added there, the maintainenance cost is null,  even if `.DS_Store` would disappear from macos in the future...
Even if there is no claim to maintain a gitignore for every OS or editor in use, I can see only benefits to covering a large portion of users.

**Comment by garciagenrique** [comment]
I just realised that you are also adding a `.gitignore` within your PR #1  @maxnoe. I don't mind closing this PR and letting you add it, if you prefer/if it will simplify things.

**Pull request closed without merging**